### ISSUE TITLE
Smart merging with wildcard import

### ISSUE THREAD

**Issue by danslapman** [liancheng/scalafix-organize-imports#105]
It'll be nice to remove redundant granular imports in such situation:

```scala
import a.b.ClassA
import a.b._
```

now it will be grouped into
```scala
import a.b.{ClassA, _}
```

but, in fact, `ClassA` is redundant here, and it could be simply

```scala
import a.b._
```

**Comment by liancheng** [comment]
Hey @danslapman, thanks for the suggestion However, quite unintuitively, this is not always safe to do. Here's an example:

```scala
import scala.collection.immutable.Set
import scala.collection.immutable._
import scala.collection.mutable._

object Main {
  val s: Set[Int] = ???
}
```

The type of `Main.s` is definitive because explicitly imported names have a higer precedence than names imported via `_`. If we merge the first two import statements into `import scala.collection.immutable._`, you get a compilation error due to ambiguity because both the mutable and immutable `Set`s are imported via `_` and have the same precedence.

The `coalesceToWildcardImportThreshold` option has a similar issue as documented in the [README][1].

[1]: https://github.com/liancheng/scalafix-organize-imports/tree/80f340cf5dfaaa0ee50e24151fee483b17df3acd#22-coalescetowildcardimportthreshold

**Comment by liancheng** [comment]
@danslapman, I added an `AggressiveMerge` option to the `groupedImports` configuration to address this issue. Please check the [updated documentation][1] for details.

[1]: https://github.com/liancheng/scalafix-organize-imports/pull/119/files#diff-1194561cbea346acb1bea37d81fdf5c70def982dcbb0c64eca2a96bc51941124R430-R486

**Comment by danslapman** [comment]
@liancheng thank you! Will check this out today

**Comment by danslapman** [comment]
Hmm I see it's not released yet

**Comment by liancheng** [comment]
@danslapman, just published 0.4.3, please have a try :)

**Comment by danslapman** [comment]
@liancheng thank you, works like a charm 👍
